plugins/importexport/csv/CSVImportExportPlugin.inc.php

```diff
@@ -93,179 +102,328 @@ function executeCLI($scriptName, &$args) {
+				if (!$authorGroup) {
+					echo __('plugins.importexport.csv.noAuthorGroup', array('press' => $pressPath)) . "\n";
+					continue;
+				}
+
+				$submission = $submissionDao->newDataObject();
+				$submission->setContextId($press->getId());
+				$submission->stampLastActivity();
+				$submission->stampModified();
+				$submission->setStatus(STATUS_PUBLISHED);
+				$submission->setWorkType($isEditedVolume == 1 ? WORK_TYPE_EDITED_VOLUME : WORK_TYPE_AUTHORED_WORK);
+				$submission->setCopyrightNotice($press->getLocalizedSetting('copyrightNotice'), $locale);
+				$submission->setLocale($locale);
+				$submission->setStageId(WORKFLOW_STAGE_ID_PRODUCTION);
+				$submission->setSubmissionProgress('');
+				$submission->setTitle($title, $locale);
+				$submission->setData('abstract', $abstract, $locale);
+
+				// $submission->setUserId($user->getId()); // REVIEW
+
+				$series = $seriesPath ? $seriesDao->getByPath($seriesPath, $press->getId()) : null;
+				if ($series) {
+					$submission->setSeriesId($series->getId());
+				}
+
+				$submissionId = $submissionDao->insertObject($submission);
+
+				$publication = $publicationDao->newDataObject();
+				$publication->setData('submissionId', $submissionId);
+				$publication->setData('version', 1);
+				$publication->setData('status', STATUS_PUBLISHED);
+				$publication->setData('datePublished', Core::getCurrentDate());
+				$publicationId = $publicationDao->insertObject($publication);
+
+				$submission->setData('currentPublicationId', $publicationId);
+				$submissionDao->updateObject($submission);
+
+				$contactEmail = $press->getContactEmail();
+				$authorString = trim($authorString, '"'); // remove double quotes if present.
+				$authors = preg_split('/;/', $authorString);
+				$firstAuthor = true;
+
+				foreach ($authors as $authorString) {
+					// Examine the author string. Best case is: Given1 Family1 <email@address.com>, Given2 Family2 <email@address.com>, etc
+					// But default to press email address based on press path if not present.
+					$givenName = $familyName = $emailAddress = null;
+					$authorString = trim($authorString); // whitespace.
+					if (preg_match('/^(\w+)(\s+\w+)?\s*(<([^>]+)>)?$/', $authorString, $matches)) {
+						$givenName = $matches[1]; // Mandatory
+						if (count($matches) > 2) {
+							$familyName = $matches[2];
+						}
+						if (count($matches) == 5) {
+							$emailAddress = $matches[4];
 						} else {
-							echo __('plugins.importexport.csv.noSeries', array('seriesPath' => $seriesPath)) . "\n";
+							$emailAddress = $contactEmail;
 						}
+					}
+					$author = $authorDao->newDataObject();
+					$author->setSubmissionId($submissionId);
+					$author->setUserGroupId($authorGroup->getId());
+					$author->setGivenName($givenName, $locale);
+					$author->setFamilyName($familyName, $locale);
+					$author->setEmail($emailAddress);
+					if ($firstAuthor) {
+						$author->setPrimaryContact(1);
```

Comment:
In my testing, this isn't properly setting any author as the primary contact on the publication.